### PR TITLE
RSDK-7402: Decide on what to do with default fake Arm

### PR DESCRIPTION
**Pull request by martha-johnston**
using fake_model.json, but there is also dofbot.json in the fake directory so I can switch it to that if wanted

fake_model.json:
Screenshot 2024-05-28 at 12 24 16 PM

dofbot.json:
Screenshot 2024-05-28 at 12 23 51 PM

### CODE REVIEW

File: components/arm/fake/fake.go [hunk 1]
```diff
@@ -81,8 +85,8 @@ func buildModel(cfg resource.Config, newConf *Config) (referenceframe.Model, err
 	case modelPath != "":
 		model, err = modelFromPath(modelPath, cfg.Name)
 	default:
-		// if no arm model is specified, we return an empty arm with 0 dof and 0 spatial transformation
-		model = referenceframe.NewSimpleModel(cfg.Name)
+		// if no arm model is specified, we return a fake arm with 1 dof and 0 spatial transformation
+		model, err = modelFromName(Model.Name, cfg.Name)
```
Comment (randhid) [inline components/arm/fake/fake.go:89]: @raybjork I'm okay with just showing a 1dof arms and adding a warning if the tests still all pass.

@martha-johnston, can you also add the dofbot, might as well while we're here/.

Comment (martha-johnston) [reply components/arm/fake/fake.go:89]: what do you mean also add the dofbot? like if a user specifically adds dofbot as the arm-model in the fake config?

Comment (martha-johnston) [reply components/arm/fake/fake.go:94]: added, let me know if that's what you meant

File: components/arm/fake/fake.go [hunk 2]
```diff
@@ -27,6 +28,14 @@ var errAttrCfgPopulation = errors.New("can only populate either ArmModel or Mode
 // Model is the name used to refer to the fake arm model.
 var Model = resource.DefaultModelFamily.WithModel("fake")
 
+var dofbotModel = "dofbot"
```
Comment (randhid) [inline components/arm/fake/fake.go:31]:
```suggestion
var dofbotModel = "yahboom-dofbot"
```

File: components/arm/fake/fake.go [hunk 3]
```diff
@@ -238,6 +254,10 @@ func modelFromName(model, name string) (referenceframe.Model, error) {
 		return ur.MakeModelFrame(name)
 	case eva.Model.Name:
 		return eva.MakeModelFrame(name)
+	case dofbotModel:
+		return makeModelFrame(model, name)
```
Comment (randhid) [inline components/arm/fake/fake.go:258]:
```suggestion
		return referenceframe.UnmarshalModelJSON(dofbotjson, name)
```

File: components/arm/fake/fake.go [hunk 4]
```diff
@@ -238,6 +254,10 @@ func modelFromName(model, name string) (referenceframe.Model, error) {
...
+	case Model.Name:
+		return makeModelFrame(model, name)
```
Comment (randhid) [inline components/arm/fake/fake.go:260]:
```suggestion
		return makeModelFrame(fakejson, name)
```

File: components/arm/fake/fake.go [hunk 5]
```diff
@@ -230,6 +239,13 @@ func (a *Arm) Geometries(ctx context.Context, extra map[string]interface{}) ([]s
 	return gif.Geometries(), nil
 }
 
+func makeModelFrame(model, name string) (referenceframe.Model, error) {
+	if model == dofbotModel {
+		return referenceframe.UnmarshalModelJSON(dofbotjson, name)
+	}
+	return referenceframe.UnmarshalModelJSON(fakejson, name)
+}
+
```
Comment (randhid) [inline components/arm/fake/fake.go:248]: no need for this function.

File: components/arm/fake/fake.go (hunk 1 above)
Comment (raybjork) [reply components/arm/fake/fake.go:94]: I think whatever we want to make the default is probably fine.

Adding the Dofbot seems ok for now but we're going to have to remove all this logic when we take these drivers out of RDK

Comment (randhid) [reply components/arm/fake/fake.go:94]: the dofbot is already out of the rdk, it only exists as a kinematics file in fake arm, it has no driver or registered resource.

**Comment by randhid** [comment]
cc @npentrel, this will now be the behavior when no model is supplied; martha could you give the fake arm doc a look and see if there are doc follow-ups and make tickets?

**Comment by randhid** [comment]
ah noticed something @martha-johnston, can we now return an error instead of logging in the constructor?

File: components/arm/fake/fake.go [hunk 6]
```diff
@@ -113,7 +122,7 @@ func (a *Arm) Reconfigure(ctx context.Context, deps resource.Dependencies, conf
 
 	dof := len(model.DoF())
 	if dof == 0 {
-		a.logger.Info("fake arm built with zero degrees-of-freedom, nothing will show up on the Control tab " +
+		return errors.New("fake arm built with zero degrees-of-freedom, nothing will show up on the Control tab " +
```
Comment (randhid) [inline components/arm/fake/fake.go:125]: let's also expand the fake arm tests to check this reconfigured behavior and test one last time with the RC card.

Comment (martha-johnston) [reply components/arm/fake/fake.go:125]: in trying to add a test I'm realizing this code is never reached. I think for there to be 0 dof, the model must be invalid so reconfigure exits with an error at an earlier point

Comment (martha-johnston) [reply components/arm/fake/fake.go:125]: wait no, that's only true for `modelFromName`

Comment (randhid) [reply components/arm/fake/fake.go:125]: if you add a kinematics file with a zero dof arm, you can reach this code,

you can add an empty kinematics json to test this.

File: components/arm/fake/zero_model.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+                    "x": 0.0,
+                    "y": 0.0,
+                    "z": 1.0,
+                    "th": 0.0
+                }
+            }
+        }
+    ],
+    "joints": []
+}
```
Comment (randhid) [inline components/arm/fake/zero_model.json:24]:
```suggestion

}
```

**Review by randhid — APPROVED** [review]

File: components/arm/fake/zero_model.json (hunk 7 above)
Comment (martha-johnston) [reply components/arm/fake/zero_model.json:24]: is this supposed to be an extra line before or after the close bracket?

Comment (randhid) [reply components/arm/fake/zero_model.json:24]: nice catch, after.

Comment (randhid) [inline components/arm/fake/zero_model.json:24]:
```suggestion
}

```

**Pull request merged**